borg/key.py

```diff
@@ -396,25 +404,43 @@ class KeyfileKey(KeyfileKeyBase):
     TYPE = 0x00
     FILE_ID = 'BORG_KEY'
 
+    def sanity_check(self, filename, id):
+        with open(filename, 'r') as fd:
+            line = fd.readline().strip()
+            if not line.startswith(self.FILE_ID):
+                raise KeyfileInvalidError(self.repository._location.canonical_path(), filename)
+            elif line[len(self.FILE_ID) + 1:] != id:
+                return KeyfileMismatchError(self.repository._location.canonical_path(), filename)
+            else:
+                return filename
+
     def find_key(self):
         id = hexlify(self.repository.id).decode('ascii')
-        keys_dir = get_keys_dir()
-        for name in os.listdir(keys_dir):
-            filename = os.path.join(keys_dir, name)
-            with open(filename, 'r') as fd:
-                line = fd.readline().strip()
-                if line.startswith(self.FILE_ID) and line[len(self.FILE_ID) + 1:] == id:
-                    return filename
-        raise KeyfileNotFoundError(self.repository._location.canonical_path(), get_keys_dir())
+        keyfile = os.environ.get('BORG_KEY_FILENAME')
+        if keyfile:
+            return satiny_check(keyfile, id)
```

Comment:
typo

and likely: missing test coverage or it would have been discovered there.